request.go

```diff
@@ -0,0 +1,102 @@
+package capnp
+
+import (
+	"context"
+	"errors"
+)
+
+// A Request is a method call to be sent. Create one with NewReqeust, and send it with
+// Request.Send().
+type Request struct {
+	method          Method
+	args            Struct
+	client          Client
+	releaseResponse ReleaseFunc
+	future          *Future
+}
+
+// NewRequest creates a new request calling the specified method on the specified client.
+// argsSize is the size of the arguments struct.
+func NewRequest(client Client, method Method, argsSize ObjectSize) (*Request, error) {
+	_, seg := NewMultiSegmentMessage(nil)
+	args, err := NewStruct(seg, argsSize)
+	if err != nil {
+		return nil, err
+	}
+	return &Request{
+		method: method,
+		args:   args,
+		client: client,
+	}, nil
+}
+
+// Args returns the arguments struct for this request. The arguments must not
+// be accessed after the request is sent.
+func (r *Request) Args() Struct {
+	return r.args
+}
+
+func (r *Request) getSend() Send {
+	return Send{
+		Method: r.method,
+		PlaceArgs: func(args Struct) error {
+			err := args.CopyFrom(r.args)
+			r.releaseArgs()
+			return err
+		},
+		ArgsSize: r.args.Size(),
+	}
+}
+
+// Send sends the request, returning a future for its results.
+func (r *Request) Send(ctx context.Context) *Future {
+	if r.future != nil {
+		return ErrorAnswer(r.method, errors.New("Sent the same request twice.")).Future()
```

Comment:
We shouldn't capitalize or punctuate error messages.

Reply:
Done.